commit 2181bfb88fb2ef13d45b3c6add677186f26f6f1b
Author: Satish Balay <balay@mcs.anl.gov>
Date:   Fri Feb 1 16:56:53 2019 -0600

    compile fixes for petsc master [as of jan 2019]

diff --git a/src/pflotran/criticality.F90 b/src/pflotran/criticality.F90
index 2b91804d8..f33e3face 100644
--- a/src/pflotran/criticality.F90
+++ b/src/pflotran/criticality.F90
@@ -9,6 +9,9 @@ module Criticality_module
 ! to waste form criticality.
 ! ===========================================================================
 #include "petsc/finclude/petscsys.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscsys
   use PFLOTRAN_Constants_module
   use Data_Mediator_Vec_class
diff --git a/src/pflotran/factory_surfsubsurface.F90 b/src/pflotran/factory_surfsubsurface.F90
index e80026350..e49e90058 100644
--- a/src/pflotran/factory_surfsubsurface.F90
+++ b/src/pflotran/factory_surfsubsurface.F90
@@ -1,6 +1,9 @@
 module Factory_Surf_Subsurf_module
 
 #include "petsc/finclude/petscsys.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscsys
   use Simulation_Surf_Subsurf_class
 
diff --git a/src/pflotran/geomech_grid.F90 b/src/pflotran/geomech_grid.F90
index c464ee36e..5b0dd0098 100644
--- a/src/pflotran/geomech_grid.F90
+++ b/src/pflotran/geomech_grid.F90
@@ -1,6 +1,9 @@
 module Geomechanics_Grid_module
 
 #include "petsc/finclude/petscvec.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscvec
   use Geomechanics_Grid_Aux_module
   use Grid_Unstructured_Cell_module
diff --git a/src/pflotran/geomech_grid_aux.F90 b/src/pflotran/geomech_grid_aux.F90
index 45598342e..081e775da 100644
--- a/src/pflotran/geomech_grid_aux.F90
+++ b/src/pflotran/geomech_grid_aux.F90
@@ -1,6 +1,9 @@
 module Geomechanics_Grid_Aux_module
 
 #include "petsc/finclude/petscvec.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscvec
   use Grid_Unstructured_Cell_module
   use Gauss_module
diff --git a/src/pflotran/geomechanics_realization.F90 b/src/pflotran/geomechanics_realization.F90
index c0d0be676..ef194f8a0 100644
--- a/src/pflotran/geomechanics_realization.F90
+++ b/src/pflotran/geomechanics_realization.F90
@@ -1,6 +1,9 @@
 module Geomechanics_Realization_class
 
 #include "petsc/finclude/petscsys.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscsys
   use Realization_Base_class
   use Geomechanics_Discretization_module
diff --git a/src/pflotran/geomechanics_regression.F90 b/src/pflotran/geomechanics_regression.F90
index c7e985be9..e5052b4f4 100644
--- a/src/pflotran/geomechanics_regression.F90
+++ b/src/pflotran/geomechanics_regression.F90
@@ -1,6 +1,9 @@
 module Geomechanics_Regression_module
  
 #include "petsc/finclude/petscvec.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscvec
   use Output_Aux_module
   
diff --git a/src/pflotran/grid.F90 b/src/pflotran/grid.F90
index a68c0cc47..9021866f3 100644
--- a/src/pflotran/grid.F90
+++ b/src/pflotran/grid.F90
@@ -1,6 +1,9 @@
 module Grid_module
 
 #include "petsc/finclude/petscmat.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscmat
   use Grid_Structured_module
   use Grid_Unstructured_module
diff --git a/src/pflotran/grid_structured.F90 b/src/pflotran/grid_structured.F90
index d36e2d186..7de78aad4 100644
--- a/src/pflotran/grid_structured.F90
+++ b/src/pflotran/grid_structured.F90
@@ -1,6 +1,9 @@
 module Grid_Structured_module
 
 #include "petsc/finclude/petscsys.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscsys
   use PFLOTRAN_Constants_module
   use Utility_module, only : Equal
diff --git a/src/pflotran/grid_unstructured.F90 b/src/pflotran/grid_unstructured.F90
index 43f52cb88..63b8cddf1 100644
--- a/src/pflotran/grid_unstructured.F90
+++ b/src/pflotran/grid_unstructured.F90
@@ -1,6 +1,9 @@
 module Grid_Unstructured_module
 
 #include "petsc/finclude/petscvec.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscvec
   use Connection_module
   use Grid_Unstructured_Aux_module
diff --git a/src/pflotran/grid_unstructured_aux.F90 b/src/pflotran/grid_unstructured_aux.F90
index f856eb70f..779de3c6b 100644
--- a/src/pflotran/grid_unstructured_aux.F90
+++ b/src/pflotran/grid_unstructured_aux.F90
@@ -2,6 +2,9 @@ module Grid_Unstructured_Aux_module
 
 !  use Connection_module
 #include "petsc/finclude/petscvec.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscvec
   use Grid_Unstructured_Cell_module
   use Geometry_module
diff --git a/src/pflotran/grid_unstructured_explicit.F90 b/src/pflotran/grid_unstructured_explicit.F90
index b46f58480..25d78f449 100644
--- a/src/pflotran/grid_unstructured_explicit.F90
+++ b/src/pflotran/grid_unstructured_explicit.F90
@@ -1,6 +1,9 @@
 module Grid_Unstructured_Explicit_module
   
 #include "petsc/finclude/petscvec.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscvec
   use Geometry_module
   use Grid_Unstructured_Aux_module
diff --git a/src/pflotran/grid_unstructured_polyhedra.F90 b/src/pflotran/grid_unstructured_polyhedra.F90
index c35e1710b..a51a0bac4 100644
--- a/src/pflotran/grid_unstructured_polyhedra.F90
+++ b/src/pflotran/grid_unstructured_polyhedra.F90
@@ -1,6 +1,9 @@
 module Grid_Unstructured_Polyhedra_module
 
 #include "petsc/finclude/petscvec.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscvec
   use Geometry_module
   use Grid_Unstructured_Aux_module
diff --git a/src/pflotran/patch.F90 b/src/pflotran/patch.F90
index ee0e8a676..7dcd029ec 100644
--- a/src/pflotran/patch.F90
+++ b/src/pflotran/patch.F90
@@ -1,6 +1,9 @@
 module Patch_module
 
 #include "petsc/finclude/petscsys.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscsys
   use Option_module
   use Grid_module
diff --git a/src/pflotran/pm_waste_form.F90 b/src/pflotran/pm_waste_form.F90
index 9ec288d0e..eb4899b07 100644
--- a/src/pflotran/pm_waste_form.F90
+++ b/src/pflotran/pm_waste_form.F90
@@ -11,6 +11,9 @@ module PM_Waste_Form_class
 ! ===========================================================================
 
 #include "petsc/finclude/petscsys.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscsys
   use PM_Base_class
   use Realization_Subsurface_class
diff --git a/src/pflotran/realization_surface.F90 b/src/pflotran/realization_surface.F90
index 453dedfb5..21ae27ff7 100644
--- a/src/pflotran/realization_surface.F90
+++ b/src/pflotran/realization_surface.F90
@@ -26,6 +26,9 @@ private
 
 
 #include "petsc/finclude/petscsys.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
 
   PetscReal, parameter :: eps       = 1.D-8
 
diff --git a/src/pflotran/regression.F90 b/src/pflotran/regression.F90
index c27d270cf..27c88d514 100644
--- a/src/pflotran/regression.F90
+++ b/src/pflotran/regression.F90
@@ -1,6 +1,9 @@
 module Regression_module
  
 #include "petsc/finclude/petscvec.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define VecScatterCreate VecScatterCreateWithData
+#endif
   use petscvec
   use Output_Aux_module
   
diff --git a/src/pflotran/solver.F90 b/src/pflotran/solver.F90
index 29908ae47..39a6fb740 100755
--- a/src/pflotran/solver.F90
+++ b/src/pflotran/solver.F90
@@ -1,6 +1,10 @@
 module Solver_module
  
 #include "petsc/finclude/petscts.h"
+#if PETSC_VERSION_GE(3,11,0)
+#define KSP_DIVERGED_PCSETUP_FAILED KSP_DIVERGED_PC_FAILED
+#define PCGetSetUpFailedReason PCGetFailedReason
+#endif
   use petscts
   use PFLOTRAN_Constants_module
   use CPR_Preconditioner_module
